server/index.js

```diff
@@ -1,6 +1,19 @@
+const socketIO = require('socket.io');
+const http = require('http');
+
 const app = require('./app');
 
 const port = app.get('port');
 
+const server = http.createServer(app);
+
+const io = socketIO(server);
+io.on('connection', (socket) => {
+  socket.on('add appointment', (value) => {
+    if (new Date(value).toDateString() === new Date().toDateString()) {
```

Comment:
this condition although it is not that important here, is easy to bypass 
but for other socket requests, you should make the io route to what the normal route do with the same amount of validation